client.go

```diff
@@ -346,3 +388,127 @@ func (c *Client) UpdateSegment(segment *Segment) error {
 
 	return nil
 }
+
+func (c *Client) GetEnvironment(apiKey string) (*Environment, error) {
+	url := fmt.Sprintf("%s/environments/%s/", c.baseURL, apiKey)
+	environment := Environment{}
+	resp, err := c.client.R().
+		SetResult(&environment).Get(url)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if !resp.IsSuccess() {
+		return nil, fmt.Errorf("flagsmithapi: Error getting environment: %s", resp)
+	}
+
+	return &environment, nil
+}
+
+func (c *Client) GetFeatureSegmentByID(featureSegmentID int64) (*FeatureSegment, error) {
+	url := fmt.Sprintf("%s/features/feature-segments/%d/", c.baseURL, featureSegmentID)
+	featureSegment := FeatureSegment{}
+	resp, err := c.client.R().
+		SetResult(&featureSegment).Get(url)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if !resp.IsSuccess() {
+		return nil, fmt.Errorf("flagsmithapi: Error getting feature segment: %s", resp)
+	}
+	return &featureSegment, nil
+}
+
+func (c *Client) UpdateFeatureSegmentPriority(featureSegmentID, priority int64) error {
+	body := []struct {
```

Comment:
Does this work if you only provide the priority of a single feature segment? I thought you had to send the whole list?

Reply:
Nope, it's not required to send the whole list. It does work with one instance as well